libsolidity/formal/CHC.cpp

```diff
@@ -634,22 +594,96 @@ void CHC::makeArrayPopVerificationTarget(FunctionCall const& _arrayPop)
 	auto previousError = m_error.currentValue();
 	m_error.increaseIndex();
 
-	addArrayPopVerificationTarget(&_arrayPop, m_error.currentValue());
+	unsigned errorId = m_context.newSlackId();
+	m_errorIds.emplace(_arrayPop.id(), errorId);
+	addVerificationTarget(VerificationTarget::Type::PopEmptyArray, &_arrayPop, m_error.currentValue());
 	connectBlocks(
 		m_currentBlock,
 		m_currentFunction->isConstructor() ? summary(*m_currentContract) : summary(*m_currentFunction),
-		currentPathConditions() && symbArray->length() <= 0 && m_error.currentValue() == static_cast<size_t>(_arrayPop.id())
+		currentPathConditions() && symbArray->length() <= 0 && m_error.currentValue() == errorId
+	);
+
+	m_context.addAssertion(m_error.currentValue() == previousError);
+}
+
+pair<smtutil::Expression, smtutil::Expression> CHC::arithmeticOperation(
+	Token _op,
+	smtutil::Expression const& _left,
+	smtutil::Expression const& _right,
+	TypePointer const& _commonType,
+	frontend::Expression const& _expression
+)
+{
+	auto values = SMTEncoder::arithmeticOperation(_op, _left, _right, _commonType, _expression);
+
+	IntegerType const* intType = nullptr;
+	if (auto const* type = dynamic_cast<IntegerType const*>(_commonType))
+		intType = type;
+	else
+		intType = TypeProvider::uint256();
+
+	// Mod does not need underflow/overflow checks.
+	// Div only needs overflow check for signed types.
+	if (_op == Token::Mod || (_op == Token::Div && !intType->isSigned()))
+		return values;
+
+	auto previousError = m_error.currentValue();
+	m_error.increaseIndex();
+
+	VerificationTarget::Type targetType;
+	unsigned errorId = m_context.newSlackId();
```

Comment:
That's a bit abusive, isn't it :-)?
Maybe ``newSlackId`` should be renamed to a more general ``newUniqueId``?

Reply:
Makes sense ;)

Reply:
Done